CourseSearchBuilderBlobTrigger/search.py

```diff
@@ -172,3 +183,140 @@ def bulk_create_courses(self, documents):
             )
 
             raise exceptions.StopEtlPipelineErrorException()
+
+
+class Synonym:
```

Comment:
Perhaps using SynonyMap rather than Synonym would be a slightly more helpful name as it seems its responsibility is creating  a synonym map and uploading it to the search service?